arrows/vxl/average_frames.cxx

```diff
@@ -88,17 +90,19 @@ template < typename PixType >
 class cumulative_frame_averager : public online_frame_averager< PixType >
 {
 public:
-  cumulative_frame_averager( bool const should_round = false );
+  cumulative_frame_averager( bool const normalize_by_n_frames = false,
```

Comment:
I'm not sure the default `false` makes sense here. I'm further not sure this should be defaulted at all.

That said... while I understand why you chose this approach, I wonder if it's worth it. It would not be much more code to implement a `sum_frame_averager` (`frame_summation`?) whose `reset` creates a zeroed image and just use `vil_sum` always. (Or have `reset` make the image have zero size and use that as the trigger to just do a copy, rather than needing a separate flag.)

Reply:
I agree with this suggestion. From a code clarity perspective it would be better to have a separate class for the sum case that always does the sum.  It fits the pattern here better and does require you to remember what the `false` means when you see `cumulative_frame_averager< PixType >{ false, round };`

That said, I'm willing to just merge this as is if it would help move things along.